### PR TITLE
Add clash warning for adding deadlines

### LINKED ISSUE #114: Clash alert for overlapping events and/or lessons and deadline reminders
if >2 events or an event(s) and lesson(s) have overlapping start and/or end times, do not allow user to add event (because scheduled lessons should take higher precedence. for deadlines that clash with lessons/events/one another, print out a reminder reminding them of which other deadlines they have on the same date and time but allow them to add it into their list.

### PR DESCRIPTION
**Pull request by honglinshang**
Fix #114

### CODE REVIEW

File: src/main/java/seedu/apollo/utils/DayTypeUtil.java [hunk 1]
```diff
@@ -58,4 +58,31 @@ public static int determineDay(String day) {
         }
     }
 
+    /**
+     * Converts a day string to an integer.
+     *
+     * @param day The string to be converted.
+     * @return The integer representation of the day string.
+     */
+    public static int determineDay(DayOfWeek day) {
```
Comment (irving11119) [inline src/main/java/seedu/apollo/utils/DayTypeUtil.java:67]: Good update of the functions utilising enums!

File: src/main/java/seedu/apollo/command/task/AddCommand.java [hunk 2]
```diff
@@ -180,13 +185,29 @@ private void addTask(TaskList taskList, Ui ui)
...
+    /**
+     * Prints warning message to user with all tasks and lessons that clash with the deadline.
+     * Will not print anything if there are no clashes.
+     *
+     * @param ui For printing warning message.
+     * @param taskList Existing tasks.
+     * @param calendar Existing lessons.
+     * @param by Due date of the deadline being added.
+     */
+    public void warnDeadlineClash(Ui ui, TaskList taskList, Calendar calendar, LocalDateTime by) {
```
Comment (irving11119) [inline src/main/java/seedu/apollo/command/task/AddCommand.java:197]: Good abstraction of the methods!

**Review by irving11119 — APPROVED** [review]

File: src/main/java/seedu/apollo/command/task/AddCommand.java [hunk 3]
```diff
@@ -180,13 +185,29 @@ private void addTask(TaskList taskList, Ui ui)
         }
     }
 
+    /**
+     * Prints warning message to user with all tasks and lessons that clash with the deadline.
+     * Will not print anything if there are no clashes.
```
Comment (yixuann02) [inline src/main/java/seedu/apollo/command/task/AddCommand.java:190]: Clear documentation, very good.

File: src/main/java/seedu/apollo/utils/DayTypeUtil.java (hunk 1 above)
Comment (yixuann02) [inline src/main/java/seedu/apollo/utils/DayTypeUtil.java:67]: Good use of enums.

**Review by yixuann02 — APPROVED** [review]
Good job adding the warnings

**Pull request merged**